src/components/Dialog/Dialog.scss

```diff
@@ -7,113 +7,194 @@
 // --------------------------------------------------
 // Dialog styles
 
+$Dialog-lgHeader-backgroundColor: $ms-color-themePrimary;
+$Dialog-default-min-width: 288px;
+$Dialog-default-max-width: 340px;
+
+// Mixin for IE9 specific styles
+@mixin dialogPositioningIE9Fallback {
+  vertical-align: middle;
+  display: inline-block;
+}
+
 .ms-Dialog {
   @include ms-baseFont;
-  @include drop-shadow(0, 0, 5px, 0, .4);
+  background-color: transparent;
+  position: fixed;
+  height: 100%;
+  width:  100%;
+  top:  0;
+  @include left(0);
+
+  // Fallback for IE9
+  display: block;
+  font-size: 0;
+  line-height: 100vh;
+  text-align: center;
+
+  // Flexbox for Modern Browsers
+  @include flexBox();
+  @include alignItems(center);
+
+  &::before {
+    @include dialogPositioningIE9Fallback();
+    content: "";
+    height: 100%;
+    width: 0;
+  }
+
+  .ms-Button.ms-Button--compound {
+    display: block;
+    @include margin-left(0);
+  }
+
+  .ms-Overlay {
+    @media screen and (-ms-high-contrast: active) {
+      opacity: 0;
+    }
+  }
+}
+
+// The actual dialog element
+.ms-Dialog-main {
+  @include dialogPositioningIE9Fallback();
+  @include drop-shadow();
   background-color: $ms-color-white;
+  box-sizing: border-box;
+  line-height: 1.35;
+  margin: auto;
+  width: $Dialog-default-min-width;
+  position: relative;
+  @include text-align(left);
+  outline: 3px solid transparent;
+}
+
+// Close button, hidden by default
+.ms-Dialog-button.ms-Dialog-button--close {
   display: none;
-  height: auto;
-  min-width: 220px;
-  max-width: 340px;
-  padding: 28px 24px;
-  z-index: $ms-zIndex-front;
-  position: absolute;
-  transform: translate(-50%, -50%);
-  left: 50%;
-  top: 50%;
+  @include margin-right(8px);
+  padding: 8px;
+
+  .ms-Icon.ms-Icon--Cancel {
+    color: $ms-color-neutralSecondary;
+    font-size: 16px;
+  }
 }
 
-.ms-Dialog.is-open {
-  display: block;
+.ms-Dialog-inner {
+  height: 100%;
+  padding: 0 28px 20px;
+}
+
+.ms-Dialog-header {
+  position: relative;
+  display: table-row;
+  width: 100%;
+  box-sizing: border-box;
 }
 
 .ms-Dialog-title {
-  font-size: $ms-font-size-xl;
-  font-weight: $ms-font-weight-light;
-  margin-bottom: 24px;
+  margin: 0;
+  display: table-cell;
+  @include ms-font-xl;
+  padding: 20px 28px 20px;
+  width: 100%;
+}
+
+.ms-Dialog-topButton {
+  display: table-cell;
+  padding: 1px;
 }
 
 .ms-Dialog-content {
   position: relative;
+  width: 100%;
+
+  // Add margin bottom between compound buttons
+  .ms-Button.ms-Button--compound:not(:last-child) {
```

Comment:
I realize this PR is a porting exercise so I won't block on this line, but as a note, we should strive to use simpler selectors for situations like this--using both `:not` and `:last-child` can be expensive for browsers to match against. 

For example, this would be better broken out like this:

```scss
.ms-Button.ms-Button--compound {
  margin-bottom: 20px;

  &:last-child {
    margin-bottom: 0;
  }
}
```